src/btree/bt_split.c

```diff
@@ -1180,16 +1165,13 @@ __split_internal(WT_SESSION_IMPL *session, WT_PAGE *parent, WT_PAGE *page)
 	WT_ASSERT(session, WT_INTL_INDEX_GET_SAFE(page) == pindex);
 	WT_INTL_INDEX_SET(page, replace_index);
 
+	/* The split is live. */
+	WT_WRITE_BARRIER();
```

Comment:
I didn't review the barrier code when I made that change, I just preserved it in place.

But, yes, I agree with you, and looking more closely, there are a few barrier calls that aren't needed and one that I think is incorrect.

I've created a PR #3288 with my suggested changes to the barriers in the split code, for your consideration.